Add known issue re sealwrapping (#2614)
hashicorp/web-unified-docs: content/vault/global/partials/important-changes/known-issue/sealwrap-raft-quorum-failures.mdx

### Requests for sealwrapped values can lead to raft quorum failures ((#sealwrap-raft-quorum-failures))

<!-- BEGIN: Vault:=v2.0.x -->

| Change      | Status   | Vault edition | Affected version | Fixed version
| ----------- | -------- | ------------- | ---------------- | -------------
| Known issue | Open     | Enterprise    | 2.0.0            | No

<!-- END: Vault:=v2.0.x -->

<!-- BEGIN: Vault:=v1.21.x -->

| Change      | Status   | Vault edition | Affected version | Fixed version
| ----------- | -------- | ------------- | ---------------- | -------------
| Known issue | Open     | Enterprise    | 1.21.0           | No

<!-- END: Vault:=v1.21.x -->

<!-- BEGIN: Vault:=v1.20.x -->

| Change      | Status   | Vault edition | Affected version | Fixed version
| ----------- | -------- | ------------- | ---------------- | -------------
| Known issue | Open     | Enterprise    | 1.20.0           | No

<!-- END: Vault:=v1.20.x -->

<!-- BEGIN: Vault:=v1.19.x -->

| Change      | Status   | Vault edition | Affected version | Fixed version
| ----------- | -------- | ------------- | ---------------- | -------------

> | Known issue | Open     | Enterprise    | 1.19.0           | 1.19.19

Reviewer: ```suggestion
| Known issue | Open     | Enterprise    | 1.19.0           | No
```
Sorry, I missed this one

### Requests for sealwrapped values can lead to raft quorum failures ((#sealwrap-raft-quorum-failures))

<!-- BEGIN: Vault:=v2.0.x -->

| Change      | Status   | Vault edition | Affected version | Fixed version
| ----------- | -------- | ------------- | ---------------- | -------------

> | Known issue | Closed   | Enterprise    | 2.0.0            | 2.0.3

Reviewer: ```suggestion
| Known issue | Open     | Enterprise    | 2.0.0            | No
```

<!-- END: Vault:=v2.0.x -->

<!-- BEGIN: Vault:=v1.21.x -->

| Change      | Status   | Vault edition | Affected version | Fixed version
| ----------- | -------- | ------------- | ---------------- | -------------

> | Known issue | Closed   | Enterprise    | 1.21.0           | 1.21.8

Reviewer: ```suggestion
| Known issue | Open     | Enterprise    | 1.21.0           | No
```

<!-- END: Vault:=v1.21.x -->

<!-- BEGIN: Vault:=v1.20.x -->

| Change      | Status   | Vault edition | Affected version | Fixed version
| ----------- | -------- | ------------- | ---------------- | -------------

> | Known issue | Closed   | Enterprise    | 1.20.0           | 1.20.13

Reviewer: ```suggestion
| Known issue | Open     | Enterprise    | 1.20.0           | No
```

<!-- END: Vault:=v1.20.x -->

<!-- BEGIN: Vault:=v1.19.x -->

| Change      | Status   | Vault edition | Affected version | Fixed version
| ----------- | -------- | ------------- | ---------------- | -------------

> | Known issue | Closed   | Enterprise    | 1.19.0           | 1.19.19

Reviewer: ```suggestion
| Known issue | Open     | Enterprise    | 1.19.0           | 1.19.19
```

<!-- END: Vault:=v1.19.x -->

In Vault 1.17.0 a change was made to assist in diagnosing deadlocks in the sealwrap code.
The deadlock detector adds overhead, and in extreme cases can prevent the Go scheduler
from scheduling the goroutines needed to maintain Raft quorum.  This would typically be
seen on clusters receiving a high volume of requests to sealwrapped mounts, particularly
after a restart or leadership change, since these result in Vault flushing its internal cache.

> Reads of cached sealwrapped values do not require locking.

Reviewer: ```suggestion
Vault 1.17.0 changed the sealwrap code to assist in diagnosing deadlocks. The
deadlock detector adds overhead and, in extreme cases, can prevent the Go
scheduler from scheduling the goroutines needed to maintain Raft quorum.

Reads of cached sealwrapped values do not require locking, so the Go scheduler
interference typically occurred on clusters receiving a high volume of requests
to sealwrapped mounts after a restart or leadership change due to Vault flushing
its internal cache.
```
Style correction: write in active voice.

Reviewer: It wasn't clear what "Reads of cached sealwrapped values do not require locking." added to the explanation, so I cut it. If you could elaborate on why we're calling that point out I can rework the suggestion.

Author: It relates to the previous sentence.  After a restart or a leadership change, our cache is empty, so requests that previously would've come from cache instead must be fetched from storage and seal-unwrapped.

Reviewer: Tweaked the suggestion to(hopefully?) make things clearer.

Even without quorum loss, less extreme cases can still increase request latency and
CPU and memory usage.

Symptoms: logs contain messages that start with:

```text
POTENTIAL DEADLOCK
```

which are followed by stack traces.  These do not necessarily indicate deadlocks are

> occurring, just that a goroutine had to wait over 30s to acquire a lock.

Reviewer: ```suggestion
Even without quorum loss, less extreme cases could still increase request
latency and CPU and memory usage and result in log entries of `POTENTIAL DEADLOCK`
followed by stack traces.

The log entries do not necessarily indicate a deadlock occured, just that a
goroutine had to wait over 30s to acquire a lock.
```
Style correction: write in active voice, avoid breaking sentences across code blocks

#### Recommendation

Upgrade to a fixed version (once available) that disables the deadlock detector by default.  There's no
reason to enable it (by adding the string "sealwrap" to the `detect_deadlocks` HCL configuration)

> unless support asks you to do so to diagnose a deadlock issue.

Reviewer: ````suggestion
Upgrade to a fixed version that disables the deadlock detector by default.

You can safely leave the detector disabled. If support asks you to enable the
detector to help diagnose a deadlock issue, set the `detect_deadlocks` field in
your sever configuration to "sealwrap":

```hcl
detect_deadlocks="sealwrap"
```
````
Style correction: avoid referencing future releases, avoid contractions
